app/models/active_record/sqlite/index.rb

```diff
@@ -0,0 +1,23 @@
+module ActiveRecord::SQLite::Index
+  extend ActiveSupport::Concern
+
+  included do
+    self.ignored_columns = [table_name.to_sym, :rank] # Rails parses our virtual table with these extra non-attributes.
```

Comment:
A SQLite `fts5` table like this is built on a `rowid` primary key and several columns to index. Active Record now understands these types of virtual tables but it doesn't recognize the `rowid` at all, not even as a column. (It's probably because the column isn't spelled out anywhere in the schema statements so Active Record has no column to parse there). 

So this module encapsulates all the work we need to have Active Record recognize the `rowid` column as a column and as the primary key.

Reply:
Oh yeah, and then Active Record also parses the table name and something called `rank` as valid columns, so we need to ignore those since they're internal to the `fts5` implementation.